File: Day10/part2.cpp
#include <bits/stdc++.h>

using namespace std;

void drawSquare(int position[3], int& index, int& count, vector<string>& s) {
    int row = index % 6;
    int col = index % 40;

    cout << row << ", " << col << endl;

    for (int i = 0; i < 3; i++) {
        if (col == position[i]) {
            s[row][col] = '#';
        }
    }
}

int main() {
    string line;
    ifstream myfile("input.txt");

    vector<string> directions;

    if (myfile.is_open()) {
        while (getline(myfile, line)) {
            directions.push_back(line);
        }
    }

    string s = "";
    vector<string> grid;

    for (int i = 1; i <= 240; i++) {
        s = s + '.';

        if (i % 40 == 0) {
            grid.push_back(s);
            s = "";
        }
    }

    grid.push_back(s);

    int cycle = 1;
    int X = 1;

    int index = 0;

    int position[3] = {0, 1, 2};

    for (int i = 0; i < directions.size(); i++) {
        drawSquare(position, index, X, grid);
        if (directions[i] == "noop") {
            index++;
        } else {
            int num = stoi(directions[i].substr(5, directions[i].length() - 1));
            index++;

            // cout << num << endl;

            drawSquare(position, index, X, grid);
            X = X + num;
            index++;

            // cout << X << endl;

            position[0] = X - 1;
            position[1] = X;
            position[2] = X + 1;

            // cout << position[0] << ", " << position[1] << ", " << position[2] << endl;
        }
    }

    for (string s : grid) {
        for (char c : s) {
            if (c == '.') {
                // black square
                cout << "\U00002B1B";
            } else {
                // white square
                cout << "\U00002B1C";
            }
        }
        cout << endl;
    }
}

File: Day10/part2.py
with open("/home/jpope/Advent-Of-Code-2022/Day10/input.txt") as file:
    lines = [line.rstrip() for line in file.readlines()]

def drawPixel(X, cycle, grid):
    pos = (cycle - 1) % 40
    if pos in {X - 1, X, X+ 1}:
        grid[cycle - 1] = "#"

X = 1
cycle = 0
grid = list("." * 40 * 6)

for line in lines:
    line = line.split()

    if line[0] == "noop":
        cycle += 1
        drawPixel(X, cycle, grid)

    elif line[0] == "addx":
        cycle += 1
        drawPixel(X, cycle, grid)

        cycle += 1
        drawPixel(X, cycle, grid)
        X += int(line[1])

answer = []

for i in range(0, 201, 40):
    answer.append("".join(grid[i: i + 40]))

for i in answer:
    for c in i:
        if c == '.':
            print("\U00002B1B", end='')
        else:
            print("\U00002B1C", end='')

    print("")